packages/api-fetch/src/middlewares/nonce.js

```diff
@@ -5,7 +5,10 @@ function createNonceMiddleware( nonce ) {
 		// If an 'X-WP-Nonce' header (or any case-insensitive variation
 		// thereof) was specified, no need to add a nonce header.
 		for ( const headerName in headers ) {
-			if ( headerName.toLowerCase() === 'x-wp-nonce' ) {
+			if (
+				headerName.toLowerCase() === 'x-wp-nonce' &&
+				headers[ headerName ] === middleware.nonce
```

Comment:
Is there any valid use-case where some user provides its own nonce (without relying on the middleware) causing this test to fail?

Reply:
I think yes, there is a valid, although unlikely, use-case when people pass their own nonces that would fail after this PR. As their nonces will be overwritten.

[Our docs](https://developer.wordpress.org/block-editor/packages/packages-api-fetch/#built-in-middlewares) ask users to add use our the built-in middleware to add nonces, which wouldn't be an issue, but if people added their own nonce manually, then it might create an issue.

A possible workaround is mark the added nonce as a "middleware nonce" some way or another, and only overwrite marked nonces.

Reply:
I'm trying to think about particular REST API calls where we'd need that? I'm not sure there is?

I think the fix is fine.